pulp_container/app/utils.py

```diff
@@ -221,6 +224,22 @@ def validate_manifest(content_data, media_type, digest):
             reason=f'{".".join(map(str, error.path))}: {error.message}', digest=digest
         )
 
+    manifests = content_data.get("manifests", None)
+    if manifests and not _is_manifest_size_valid(manifests):
+        raise ManifestInvalid(
+            reason="Manifest size is not valid, the max allowed size is {}.".format(
+                MANIFEST_PAYLOAD_MAX_SIZE
+            ),
+            digest=digest,
+        )
+
+
+def _is_manifest_size_valid(manifests):
+    for manifest in manifests:
+        if manifest.get("size") > MANIFEST_PAYLOAD_MAX_SIZE:
```

Comment:
i would not trust what's written on the manifest list json file other then actually checking the manifest file size